nahusenayElias/Team_4
# DRD-74: Create a custom module to fetch contact segments from Mautic API

## Related ticket
[DRD-74](https://edu-team4-react24k.atlassian.net/jira/software/projects/DRD/boards/1?selectedIssue=DRD-74)

## In this PR: 
- Created a custom module called MauticAPI to fetch data from Mautic
- Added two contrib modules: dotenv (for environmental variables like API username and the API key or password) and tac_lite (not yet used in this PR)
- Added a load.environment.php file for environmental variables
- Exported new database for Mautic

## Testing instructions
- checkout branch
- lando composer install
- lando composer cim
- lando drush cr
- Add .env file to the root with the following content: 
`APP_ENV=prod
API_USER=admin
API_KEY=+Nimda666+`
- cd mautic-lando
- lando db-import mautic-DRD-74.gz
- Check from Drupal's Extend tab that the Mautic API module has been installed and enabled
- Go to localhost:YOURDRUPALPORT/mautic/segments and check that you can see the two segments created in Mautic


[DRD-74]: https://edu-team4-react24k.atlassian.net/browse/DRD-74?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
